src/vmm/src/vstate/vcpu/x86_64.rs

```diff
@@ -433,7 +433,7 @@ impl KvmVcpu {
         // per million beacuse it is common for TSC frequency
         // to differ due to calibration at boot time.
         let diff = (i64::from(self.get_tsc_khz()?) - i64::from(state_tsc_freq)).abs();
-        Ok(diff > i64::from(state_tsc_freq * TSC_KHZ_TOL_NUMERATOR / TSC_KHZ_TOL_DENOMINATOR))
+        Ok(diff > i64::from(state_tsc_freq).checked_mul(TSC_KHZ_TOL_NUMERATOR).unwrap() / TSC_KHZ_TOL_DENOMINATOR)
```

Comment:
This should be an ok_or but I guess I need to introduce a new error for this to work... or are we happy enough that this should not overflow given that it's `u32 * 250`?

Reply:
Yeah, I'm perfectly happy to keep it with `*` since u32 * 250 cannot overflow!